pkg/provisioner/ironic/raid.go

```diff
@@ -99,11 +99,33 @@ func buildTargetHardwareRAIDCfg(volumes []metal3v1alpha1.HardwareRAIDVolume) (lo
 			}
 			nameCheckFlags[volume.Name] = index
 		}
+
+		// Check that controller field is specified if PhysicalDisks are used
+		if len(volume.PhysicalDisks) != 0 {
+			if volume.Controller == "" {
+				return nil, errors.Errorf("'controller' must be specified if 'physicalDisks' are used!")
+			}
+		}
+
+		// Check numberOfPhysicalDisks is same as len(physicalDisks)
+		if volume.NumberOfPhysicalDisks != nil && len(volume.PhysicalDisks) != 0 {
```

Comment:
Q: what if `PhysicalDisk` is defined but not `NumberOfPhysicalDisks`?

Reply:
I'd incline towards what @zaneb has said. Specifying ``NumberOfPhysicalDisks`` just doesn't need to be conflicting with the number of ``PhysicalDisks``.

Reply:
Ok thanks for the clarification